### PR TITLE
Multiple channels

### PR DESCRIPTION
**Pull request by oilnam**
1. Add support for channels. This will store every report under `s3-path/reports/$channel/$timestamp`, where `$channel` is {uk, us, au}.

2. Make the creation of the reports asynchronous

### CODE REVIEW

File: src/main/scala/com/gu/contentapi/Lambda.scala [hunk 1]
```diff
@@ -36,9 +53,9 @@ class Lambda extends LazyLogging with Retry {
     }
   }
 
-  private def uploadReport(report: String, createdAt: DateTime): Option[String] = {
+  private def uploadReport(report: String, createdAt: DateTime, channelName: String): Option[String] = {
```
Comment (mchv) [inline src/main/scala/com/gu/contentapi/Lambda.scala]:
```scala
uploadReport(report: String, createdAt: DateTime, channel: Channel)
```

File: src/main/scala/com/gu/contentapi/Lambda.scala [hunk 2]
```diff
@@ -36,9 +53,9 @@ class Lambda extends LazyLogging with Retry {
...
     logger.info("Uploading report to S3...")
-    S3.uploadToS3(report, createdAt) match {
+    S3.uploadToS3(report, createdAt, channelName) match {
```
Comment (mchv) [inline src/main/scala/com/gu/contentapi/Lambda.scala]:
```scala
channel.name
```

**Comment by mchv** [comment]
👍  As discussed we will avoid `Future` by having 3 lambda entry points (one for each report)

File: src/main/scala/com/gu/contentapi/Config.scala [hunk 3]
```diff
@@ -11,11 +11,21 @@ object Config extends LazyLogging {
 
   val appleNewsUrl = Option(conf.getProperty("apple.url")).getOrElse(sys.error("missing apple url"))
 
-  val apiKeyId = Option(conf.getProperty("apple.api.key.id.uk")).getOrElse(sys.error("missing api key id"))
+  val apiKeyIdUk = Option(conf.getProperty("apple.api.key.id.uk")).getOrElse(sys.error("missing uk api key id"))
```
Comment (LATaylor-guardian) [inline src/main/scala/com/gu/contentapi/Config.scala]: Could create a `getMandatory` function here to reduce duplicaton.

**Pull request merged**